ringersong/src/main/java/com/RingerSong/free/viewmodel/RingerViewModel.kt

```
@@ -740,4 +740,133 @@ class RingerViewModel @Inject constructor(
             onComplete(info)
         }
     }
+
+    fun addSharedUrl(url: String, onResult: (String) -> Unit) {
```

detekt (1.23.8)

[warning] The function addSharedUrl appears to be too complex based on Cognitive Complexity (complexity: 29). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)